torch/fx/passes/infra/partitioner.py

```diff
@@ -164,14 +164,13 @@ def dfs_iter_find_cycle(all_user_nodes: Set[Node]):
 
         def merge_single_node(node: Node, id: Optional[int]):
             def _update_partition_map(node: Node, id: int):
-                # Iterate through all the downstream nodes of this node and update the partition map
-                # to indicate that there is a path from the partition id of this node to the target
-                # partition id.
-                downstream_nodes = self.dependency_viewer.downstreams_of(node)
-                for curr_node in downstream_nodes:
-                    target_id = assignment.get(curr_node, None)
+                # Iterate through all the users of this node and update the partition map to indicate
+                # that there is a path from the partition id of this node to the target partition id.
+                for user_node in node.users:
+                    target_id = assignment.get(user_node, None)
```

Comment:
So in order for this to work we're assuming that the graph is being traversed in reverse. I think maybe we should add an assert that `target_id is not None` here to ensure that the user node has already been traversed?

Reply:
Done.